packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -310,12 +300,12 @@ export default createRule<Options, MessageId>({
       );
     }
     function checkCallExpression(node: TSESTree.CallExpression): void {
-      const {
-        arguments: [callback],
-      } = node;
-      if (callback && shouldCheckCallback(node)) {
+      // If this is something like arr.filter(x => /*condition*/), check `condition`
+      if (isArrayPredicateFunction(node) && node.arguments.length) {
```

Comment:
Did a minor bit of refactor, I realized when rereading this code that the original version wasn't very clear.  (Calling a variable `callback` before actually checking that it's a callback, for example)